### PR TITLE
Implement /kick

### LINKED ISSUE #992: /kick command
Example:

You:
 Input: /kick Adamx120

TFM:
 Sender - Kicking Adamx120 from the server

 Player:
 You have been kicked from the server
                 Reason: %reason

### PR DESCRIPTION
**Pull request by DUFCLiam**
/kick command

#992

### CODE REVIEW

**Comment by Wild1145** [comment]
You are currently missing the file extension... Again I have no idea how thats possible because to have tested it before submitting the PR you would have realized that it doesnt even compile... And just about every IDE automatically adds the extensions anyway if you do it right...

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+@CommandPermissions(level = AdminLevel.SUPER, source = SourceType.BOTH)
+@CommandParameters(description = "Kick someone from the server", usage = "/<command> <playername> <reason>")
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick]: Reason should display as optional

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{
+    public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+        String reason = null;
+        if (args.length >= 2)
+        {
+            reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+        }
+            {
+                if (!TFM_AdminList.isSuperAdmin(sender))
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick]: Should really check if the person that is trying to be kicked it of an equal or higher rank, if so then deny the ability. If they are a lower rank then allow them to kick.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                    sender.sendMessage(TFM_Command.MSG_NO_PERMS);
+                }
+                
+                if (!TFM_AdminList.isSuperAdmin(sender_p))
+                {
+                    if (args.length < 1) {
+                        
+                    sender_p.kickPlayer(ChatColor.RED + "You have been kicked by: " + sender.getName() + "-" + " No reason given");
+                    }
+                        }
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Formatting.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+public class Command_kick
+{
+    public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+        String reason = null;
+        if (args.length >= 2)
+        {
+            reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+        }
+            {
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Unnecessary brace open.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                    sender.sendMessage(TFM_Command.MSG_NO_PERMS);
+                }
+                
+                if (!TFM_AdminList.isSuperAdmin(sender_p))
+                {
+                    if (args.length < 1) {
+                        
+                    sender_p.kickPlayer(ChatColor.RED + "You have been kicked by: " + sender.getName() + "-" + " No reason given");
+                    }
+                        }
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick]: Formatting has gone to pot here.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+@CommandPermissions(level = AdminLevel.SUPER, source = SourceType.BOTH)
+@CommandParameters(description = "Kick someone from the server", usage = "/<command> <playername> <reason>")
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: This should still be optional... (I know you did just add the .java but thats not the point)

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+{
+    public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+        String reason = null;
+        if (args.length >= 2)
+        {
+            reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+        }
+            {
+                if (!TFM_AdminList.isSuperAdmin(sender))
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: This again should be checking if the player is of a higher rank than the command sender. If so then do not kick. If they are a lower rank then allow the kick.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                {
+                    sender.sendMessage(TFM_Command.MSG_NO_PERMS);
+                }
+                
+                if (!TFM_AdminList.isSuperAdmin(sender_p))
+                {
+                    if (args.length < 1) {
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Formatting

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                {
+                    sender.sendMessage("You cannot kick admins!");
+                }
+                
+                if (!TFM_AdminList.isSuperAdmin(sender_p))
+                
+                {
+                    sender_p.kickPlayer(ChatColor.RED + reason + "- " + sender.getName());
+                    TFM_Util.bcastMsg(sender.getName() + "-" + " Kicking" + sender_p.getName() + ChatColor.YELLOW + "- " + reason , ChatColor.RED);
+
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Random blank line... Formatting

**Comment by DUFCLiam** [comment]
@JeromSar @Wild1145 This should be fixed now.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+public class Command_kick
+{
+    public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+        String reason = null;
+        if (args.length >= 2)
+        {
+            reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+        }
+                if (!TFM_AdminList.isSuperAdmin(sender))
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Formatting.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+                }
+                else
+                {
+                    sender.sendMessage("You cannot kick admins!");
+                }
+                
+                if (!TFM_AdminList.isSuperAdmin(sender_p))
+                
+                {
+                    sender_p.kickPlayer(ChatColor.RED + reason + "- " + sender.getName());
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: You're imitating `TFM_Util.adminAction()` please use that instead.
Additionally, display the message before anything happens.

**Comment by JeromSar** [comment]
@DUFCLiam Your feature does not function.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+                if (!TFM_AdminList.isSuperAdmin(sender))
+                {
+                    sender.sendMessage(TFM_Command.MSG_NO_PERMS);
+                }
+                
+                if (!TFM_AdminList.isSuperAdmin(sender_p))
+                {
+                    if (args.length < 1) 
+                    {
+
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Random blank line...

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+        if (args.length >= 2)
+        {
+            reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+        }
+            if (!TFM_AdminList.isSuperAdmin(sender))
+            {
+                sender.sendMessage(TFM_Command.MSG_NO_PERMS);
+            }
+                
+                if (!TFM_AdminList.isSuperAdmin(sender_p))
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Formatting

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+@CommandPermissions(level = AdminLevel.SUPER, source = SourceType.BOTH)
+@CommandParameters(description = "Kick someone from the server.", usage = "/<command> <playername> [reason]")
+public class Command_kick
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: does not extend the TFM_Command file.

Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: @DUFCLiam  As stated, this PR sadly has not been tested before committing, which leads to these embarrassing oversights. Please follow the contribution guidelines from now on.

**Comment by DUFCLiam** [comment]
Fixed? @JeromSar or @Wild1145

**Comment by JeromSar** [comment]
@DUFCLiam I will remove the "needs testing" label when you have complied to the contributing guidelines and sufficiently tested your PR. This PR inherently _does not function at all_, and until it does, I'll refrain from commenting.

**Comment by DUFCLiam** [comment]
Ok, @JeromSar

**Comment by DUFCLiam** [comment]
<table>
  <tr>
    <td> lots of stuff...
      more stuff
      </td>
    <td> alternative for short lines </td>
    <td> etc. </td>
    </tr>
  </table>

<table>
  <tr> ... etc
  </table>

**Comment by ReelPattytothewatty** [comment]
Tested it,  when someone is kicked, they can still bypass it.

**Comment by Wild1145** [comment]
@ReelPattytothewatty What is there to bypass? You are simply being kicked from the server potentially with a reason...

**Comment by ReelPattytothewatty** [comment]
@Wild1145 When you do /kick <playername> then do /glist ban <playername> you can bypass the ban.

**Comment by DUFCLiam** [comment]
@ReelPattytothewatty, Show us proof of this.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+            if (!TFM_AdminList.isSuperAdmin(sender_p))
+            {
+                if (args.length < 1) 
+                {
+                sender_p.kickPlayer(ChatColor.RED + "You have been kicked by: " + sender.getName() + "-" + " No reason given");
+                }
+                   }
+                else
+                {
+                    sender.sendMessage("You cannot kick admins!");
```
Comment (rsajeey) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Use playerMsg.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+            if (!TFM_AdminList.isSuperAdmin(sender))
+            {
+                sender.sendMessage(TFM_Command.MSG_NO_PERMS);
+            }
+                
+            if (!TFM_AdminList.isSuperAdmin(sender_p))
+            {
+                if (args.length < 1) 
+                {
+                sender_p.kickPlayer(ChatColor.RED + "You have been kicked by: " + sender.getName() + "-" + " No reason given");
```
Comment (OnlineRepz) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Formatting

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+                }
+                   }
```
Comment (OnlineRepz) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_kick.java]: Formatting

**Pull request closed without merging**